### ISSUE TITLE
Publish button always available to users with node edit access

### ISSUE THREAD

**Issue by laryn** [backdrop-contrib/save_draft#19]
From the D7 queue with a RTBC patch:

> When you set at Publish Options in a content type in default options to default status = not published, save_draft module is enabled and configured to allow all authenticated users to save draft, the button publish appears to all users who have the permission to create a new content type, even when you want this new content needs to be approved before publication.

https://www.drupal.org/project/save_draft/issues/1778836

**Comment by herbdool** [comment]
@laryn I was kinda relying on this. I had seen in the code that if someone has permission to edit the node, they have permission to publish. If this PR gets incorporated you may need to make it backwards-compatible and provide a new permission and give it to all roles with permission to view an unpublished node.

**Comment by laryn** [comment]
@herbdool Yes, there would be some backwards-compatibility concerns for sure. Currently the Save Draft permission indicates:

> Allows people with permission to view and edit their own unpublished content to change the published state of their content when saving it.

**Comment by herbdool** [comment]
@laryn yeah, I'm not sure if it's worth the effort, when the description of the permission says it permits "change the published state". The original issue is concerned basically about admins (those who have permission to *create* a content type) have permission to publish content. Not sure why that's a pressing concern.
